simplic-authentication/Simplic.Authentication.Service/AuthenticationService.cs

```diff
@@ -41,45 +45,133 @@ public AuthenticationService(IUnityContainer container, IAuthorizationService au
 
-            var user = userService.GetByName(userName);
+            if (!provider.CheckCredentials(domain, userName, password, user.Password))
+                throw new LoginFailedException(LoginFailedType.InvalidCredentials);
 
+            return GenerateUserSession(user);
+        }
+
+        /// <summary>
+        /// Generate user session by current user instance
+        /// </summary>
+        /// <param name="user">User instance</param>
+        /// <returns>UserSession object if valid</returns>
+        private Session GenerateUserSession(User.User user)
+        {
             if (user == null)
                 throw new LoginFailedException(LoginFailedType.UserNotFound);
 
             if (!user.IsActive)
                 throw new LoginFailedException(LoginFailedType.UserNotActive);
 
-            var providerName = user.IsADUser ? "ActiveDirectoryCredentialProvider" : "DefaultCredentialProvider";
-            var provider = container.Resolve<ICredentialProvider>(providerName);
-
-            if (!provider.CheckCredentials(domain, userName, password, user.Password))
-                throw new LoginFailedException(LoginFailedType.InvalidCredentials);
-
             var userGroups = groupService.GetAllByUserId(user.Ident);
             var isSuperUser = userGroups.Any(x => x.GroupId == 0);
             var userGroupIdents = userGroups.Select(x => x.Ident).ToList();
 
-            session = new Session()
+            return new Session()
             {
                 UserId = user.Ident,
-                UserName = user.UserName,
-                
+                UserName = user.UserName,
+
                 IsSuperUser = isSuperUser,
                 UserAccessGroups = userGroupIdents,
                 UserAccessGroupsBitMask = authorizationService.CreateBitMask(userGroupIdents),
                 UserBitMask = authorizationService.CreateBitMask(user.Ident),
                 IsADUser = user.IsADUser
-            };
-
-            return session;
-        }
+            };
+        }
+
+        /// <summary>
+        /// Activate autologin and write autologin file
+        /// </summary>
+        /// <param name="domain">Current domain</param>
+        /// <param name="userName">Current user</param>
+        /// <param name="password">Current password</param>
+        public void SetAutologin(string domain, string userName, string password)
+        {
+            try
+            {
+                var path = GetAutologinPath();
+
+                if (Login(domain, userName, password) != null)
+                {
+                    // Create autologin object
+                    var hash = GenerateHash(userName, domain);
+                    var loginObject = new AutologinModel() { Hash = hash, UserName = userName, Domain = domain };
+
+                    // Ensure directory
+                    IO.DirectoryHelper.CreateDirectoryIfNotExists(Path.GetDirectoryName(GetAutologinPath()));

```

Comment:
instaed of calling `GetAutologinPath()` use the already assigned variable `path`